blur/src/blur-exchange.ts

```diff
@@ -0,0 +1,53 @@
+import { Address, BigInt, log } from "@graphprotocol/graph-ts"
+import { OrdersMatched } from "../generated/BlurExchange/BlurExchange"
+import * as airstack from "../modules/airstack/nft-marketplace"
+import { PROTOCOL_SELL_ACTION_TYPE, MARKET_PLACE_TYPE } from "./constants"
+import { _canMatchOrders } from "./blur-utils"
+import { BIG_INT_ZERO } from "../modules/airstack/common"
+export function handleOrdersMatched(event: OrdersMatched): void {
+    let txHash = event.transaction.hash.toHexString()
+    let buyer = event.params.buy.trader
+    let seller = event.params.sell.trader
+
+    let orderDetails = _canMatchOrders(txHash, event.params.sell, event.params.buy)
```

Comment:
As this is already validated in contract, we can rely on events to get the sale details.